runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -2363,8 +2349,230 @@ remoteCompile(
+               }
+            }
+         }
+      else
+         {
+         if (TR::Options::getAOTCmdLineOptions()->getVerboseOption(TR_VerboseJITaaS))
+            {
+            TR_VerboseLog::writeLineLocked(TR_Vlog_JITaaS, " Failed AOT cache validation");
+            }
+
+         TR::CompilationInfo::disableAOTCompilations();
+         }
+
+#if defined(J9VM_INTERP_AOT_RUNTIME_SUPPORT)
+
+      TR_Debug *debug = TR::Options::getDebug();
+      bool canRelocateMethod = false;
+      if (debug)
+         {
+         TR_FilterBST *filter = NULL;
+         J9UTF8 *className = ((TR_ResolvedJ9Method*)comp->getCurrentMethod())->_className;
+         J9UTF8 *name = ((TR_ResolvedJ9Method*)comp->getCurrentMethod())->_name;
+         J9UTF8 *signature = ((TR_ResolvedJ9Method*)comp->getCurrentMethod())->_signature;
+         char *methodSignature;
+         char arr[1024];
+         int32_t len = J9UTF8_LENGTH(className) + J9UTF8_LENGTH(name) + J9UTF8_LENGTH(signature) + 3;
+         if (len < 1024)
+            methodSignature = arr;
+         else
+            methodSignature = (char *) TR_MemoryBase::jitPersistentAlloc(len);
+
+         if (methodSignature)
+            {
+            sprintf(methodSignature, "%.*s.%.*s%.*s", J9UTF8_LENGTH(className), utf8Data(className), J9UTF8_LENGTH(name), utf8Data(name), J9UTF8_LENGTH(signature), utf8Data(signature));
+            //printf("methodSig: %s\n", methodSignature);
+
+            if (debug->methodSigCanBeRelocated(methodSignature, filter))
+               canRelocateMethod = true;
+            }
+         else
+            canRelocateMethod = true;
+
+         if (methodSignature && (len >= 1024))
+            TR_MemoryBase::jitPersistentFree(methodSignature);
+         }
+      else
+         {
+         // Prevent the relocation if specific option is given
+         if (!comp->getOption(TR_DisableDelayRelocationForAOTCompilations))
+            canRelocateMethod = false;
+         else
+            canRelocateMethod = true;
+         }
+
+      if (canRelocateMethod)
+         {
+         J9JITDataCacheHeader *cacheEntry;
+
+         TR_ASSERT_FATAL(comp->cg(), "CodeGenerator must be allocated");
+
+         // Use same code cache as AOT compile
+         //
+         TR::CodeCache *aotMCCRuntimeCodeCache = comp->cg()->getCodeCache();
+         TR_ASSERT(aotMCCRuntimeCodeCache, "Must have a reserved codeCache");
+         cacheEntry = (J9JITDataCacheHeader *)dataStart;
+
+         int32_t returnCode = 0;
+
+         if (TR::Options::getVerboseOption(TR_VerboseJITaaS))
+            {
+            TR_VerboseLog::writeLineLocked(
+               TR_Vlog_JITaaS,
+               "Applying JITaaS remote AOT relocations to newly AOT compiled body for %s @ %s",
+               comp->signature(),
+               comp->getHotnessName()
+               );
+            }
+         try
+            {
+            TR::CompilationInfoPerThreadBase::InterruptibleOperation(*entry->_compInfoPT);
```

Comment:
This line should be removed, I think.
At this point we are alredy inside Interruptible operation region, which is started in `TR::CompilationInfoPerThreadBase::compile`, so trying to enter it again triggers assert.